core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -161,6 +161,20 @@ default ConnectorTableHandle makeCompatiblePartitioning(ConnectorSession session
         return Optional.empty();
     }
 
+    /**
+     * Return table schema definition for the specified table handle.
+     * Table schema definition is a set of information
+     * required by semantic analyzer to analyze the query.
```

Comment:
Do we need to define the content of the TableSchema by referring to the engine internal?

connector implementor may not care what is semantic analyzer and what other analyzers we have internally.

if you want to refer to usecase, maybe it would be better to say that this is the information required when planning a SELECT query?

Note however, that even in such case, we call other methods, like getTableProperties